internal/core/items/component_container.rs

```diff
@@ -66,6 +69,44 @@ impl ComponentContainer {
         };
 
         let product = factory.build(factory_context);
+        if let Some(item_tree) = product.clone() {
+            let item_tree = vtable::VRc::borrow_pin(&item_tree);
+            let root_item = item_tree.as_ref().get_item_ref(0);
+            if let Some(window_item) =
+                crate::items::ItemRef::downcast_pin::<crate::items::WindowItem>(root_item)
+            {
+                // Do _not_ use a two-way binding: That causes evaluations of width and height to
+                // assert on recursive property evaluation.
+                let weak = self.self_weak.get().unwrap().clone();
+                super::WindowItem::FIELD_OFFSETS
+                    .width
+                    .set_binding(
+                        window_item,
+                        Box::new(move || {
+                            let self_rc = weak.upgrade().unwrap();
+                            let self_pin = self_rc.borrow();
+                            let self_cc = crate::items::ItemRef::downcast_pin(self_pin).unwrap();
+                            ComponentContainer::FIELD_OFFSETS.width.apply_pin(self_cc).get()
```

Comment:
We generate getters for all properties, so I think you can just write `self_cc.width()` instead of the `FIELD_OFFSETS` dance.

Reply:
I don't get it,  that item should be a `Pin<&ComponentContainer>` , right?  This is defined in that file and it derives the `SlintElement` derive macro